csrc/ops/utils.h

```diff
@@ -33,6 +33,14 @@ Val* newScalar(ValType vtype, DataType dtype);
 
 IterType promoteIterType(IterType type1, IterType type2);
 
+//! Maps the input iterdomains to the output of MatmulOp.
+std::vector<IterDomain*> mapMatmulOpIterDomains(
+    const std::vector<IterDomain*>& input_domain,
+    bool is_lhs,
+    size_t out_size);
+
+IterDomain* newOutputIterDomain(const std::vector<IterDomain*>& ids);
```

Comment:
What's this function for?

Reply:
The `newOutputDomain` takes a vector of tensorviews and assumes they are all aligned to create the output tensorview. Since in the case of matmul, the inputs are not aligned to the output and the mapping is more complex, I refactored this part of the code from `newOutputDomain` into a separate `newOutputIterDomain` that takes a vector of aligned input iterdomains to create the output iterdomain.

In `newOutputDomain` (which we use in binary ops), we simply call this function for every set of input iterdomains at any axis. For `MatmulOp`, we specifically identify which input iterdomains correspond to a particular output iterdomain and use this function instead of using `newOutputDomain`.